### PR TITLE
Release v4.2

### PR DESCRIPTION
**Pull request by marclecerf**
- [x] Add new Python unit tests to GH actions
- [x] Add PyPI wheel publish to GH actions

Minor release.

Improvements:
- Added `upsp.video` subpackage with mraw and cine readers
- Bounds checking for 3D->2D projection - ensures the internal calibration is valid to apply to a given model position (e.g., in cases where higher-order calibration terms are negative, points very far outside the field of view can get erroneously projected INTO the image plane)
- Template-based detection for registration targets (+deprecate blob detection)
- Added unit tests for camera calibration modules

Smaller updates/fixes:
- `upsp.kulite_utilities.Kulites` reduce spam
- `upsp-make-processing-tree` bugfixes
    - `--outdir` handles relative paths; they are converted to canonical paths prior to generating tree.
    - `config_name` parsing is improved.
- psp_process input (`*.inp`) file now includes optional `pixel_interpolation` parameter to control image warping pixel interpolation method(`linear` or `nearest`). Defaults to `linear` (same as previous hard-coded behavior)

### CODE REVIEW

File: test/python/test_video.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            self.assertEqual(vid.bit_depth, 12)
+            self.assertEqual(vid.frame_count, 2)
+
+            # frame reading
+            f0 = vid.read_frame(1)
+            vid.read_frames(2)
+
+            self.assertEqual(f0.shape, (1024, 1024))
+
+    def test_from_cih_path(self):
```
Comment (ixjlyons) [inline test/python/test_video.py]: For some reason I thought this behavior worked correctly, but it was never implemented. We can just remove or skip this test for now.

**Comment by marclecerf** [comment]
@ixjlyons 

Ready for review. Final behavior re: Python version support:
- pyproject.toml specifies ~=3.9.0 (only Python 3.9 + patches)
- You should see the following error when trying "pip install upsp" with a version other than 3.9:

    ```
    ERROR: Ignored the following versions that require a different python version: 4.2a2 Requires-Python ~=3.9.0
    ERROR: Could not find a version that satisfies the requirement upsp (from versions: 4.2a1)
    ERROR: No matching distribution found for upsp
    ```

NOTE: I yanked 4.2a1 on PyPI so it isn't accidentally used by pip.

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -82,33 +82,19 @@ jobs:
...
-      - name: Run build+install with pip
+      - name: Run CMake build
         run: |
-          SKBUILD_CONFIGURE_OPTIONS=\
-          " -DCMAKE_TOOLCHAIN_FILE=${VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"\
-          " -DVCPKG_TARGET_TRIPLET=x64-linux-dynamic" pip install -v .
+          mkdir build
+          cd build
+          cmake \
+            -DCMAKE_TOOLCHAIN_FILE="${VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake" \
```
Comment (ixjlyons) [inline .github/workflows/unit-tests.yml:90]: It looks like this workflow was disabled, but should this now specify `UPSP_BUILD_TESTING=ON` as well?

**Review by ixjlyons — APPROVED** [review]
Aside from the question about the unit testing workflow, looks good to me.

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (marclecerf) [reply .github/workflows/unit-tests.yml:90]: Thank you for the catch. Yes will re-enable this workflow and force-set cmake opts (now that by default they are OFF)

**Pull request merged**